src/app/cats/cats-image/cats-image.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+import { Cat } from '../model';
+import { CatsDataService } from '../cats-data.service';
+import { ActivatedRoute, ActivatedRouteSnapshot } from '@angular/router';
+
+@Component({
+  selector: 'app-cats-image',
+  templateUrl: './cats-image.component.html',
+  styleUrls: ['./cats-image.component.scss']
+})
+export class CatsImageComponent{
+
+  constructor(private catsDataService: CatsDataService, private activatedRoute: ActivatedRoute) {
+    this.route = activatedRoute.snapshot;
+  }
+  idx = 0;
+  cat: Cat;
+  adoptionFormActive: boolean;
+  showConfirmMsg: boolean;
+  private route: ActivatedRouteSnapshot;
+
+  ngOnInit(): void {
+    const catName = this.route.params['catName'];
+    this.catsDataService.getCat(catName).subscribe(data => this.cat = data);
```

Comment:
prawdopodobniej prościej by było przekazać kota jako input, skoro już mamy go w garści poziom wyżej.
To rozwiązanie ciągle jest zupełnie przyzwoite i działa dobrze.